community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/ast/convert/commands/StatementConverters.scala

```diff
@@ -230,6 +230,8 @@ object StatementConverters {
         commands.SchemaIndex(identifier.name, label.name, property.name, commands.AnyIndex, None)
       case ast.UsingScanHint(identifier, label) =>
         commands.NodeByLabel(identifier.name, label.name)
+      case ast.UsingJoinHint(identifier) =>
+        commands.NodeJoin(identifier.name)
```

Comment:
Since we are simply going to be throwing out the join hint for the rule planner, we might as well not create commands objects representing the hint, and instead just ignoring it here instead. Remember to send a notification when the hint is being dropped though